sql/core/src/main/scala/org/apache/spark/sql/catalyst/SQLBuilder.scala

```diff
@@ -356,8 +372,14 @@ class SQLBuilder(logicalPlan: LogicalPlan) extends Logging {
       }
     }
 
+    val broadcastHint = project match {
+      case p @ Project(projectList, Hint("BROADCAST", tables, child)) =>
+        if (tables.nonEmpty) s"/*+ MAPJOIN(${tables.mkString(", ")}) */" else ""
+      case _ => ""
+    }
     build(
       "SELECT",
+      broadcastHint,
```

Comment:
To do this, we need to add the extra dummy `Project` for the following condition.
```scala
case a @ Aggregate(_, _, e @ Expand(_, _, p: Project)) if isGroupingSet(a, e, p) =>
      groupingSetToSQL(a, e, p)
```